File: src/main/java/io/zeitwert/fm/task/model/enums/CodeTaskPriority.java

package io.zeitwert.fm.task.model.enums;

import io.dddrive.enums.model.base.EnumeratedBase;
import lombok.Data;
import lombok.experimental.SuperBuilder;

@Data
@SuperBuilder
public final class CodeTaskPriority extends EnumeratedBase {

	@Override
	public boolean equals(Object other) {
		return super.equals(other);
	}

	@Override
	public int hashCode() {
		return super.hashCode();
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/task/model/enums/CodeTaskPriority.kt
package io.zeitwert.fm.task.model.enums

import dddrive.ddd.enums.model.EnumeratedEnum
import dddrive.ddd.enums.model.base.EnumerationBase

enum class CodeTaskPriority(
	override val defaultName: String,
) : EnumeratedEnum {

	LOW("Tief"),
	NORMAL("Normal"),
	HIGH("Hoch"),
	;

	override val enumeration get() = Enumeration

	companion object Enumeration : EnumerationBase<CodeTaskPriority>(CodeTaskPriority::class.java) {
		init {
			entries.forEach { addItem(it) }
		}

		@JvmStatic
		fun getPriority(itemId: String?) = if (itemId != null) getItem(itemId) else null
	}
}
